src/App.js

```diff
@@ -1,63 +1,142 @@
+
+
+    try {
+      const response = await fetch(url, options);
 
 
-    useEffect(() => {
-      if (!isLoading) {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList)); //save the todoList as a string in the localStorage whenever it changes 
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
       }
-    }, [todoList, isLoading]); // execute/call the callback function(effect) only when the 'todoList' state variable changes by specifying 'todoList' as the dependency array
 
 
+      // Parse the response JSON
+      const data = await response.json();
 
 
-    function removeTodo(id) {
-      // Filter the todoList array to exclude the todo item with the specified id
-      const updatedTodoList = todoList.filter((todo) => todo.id !== id);
+      // Print the fetched data to observe Airtable's API response
+      console.log('Fetched data:', data);
 
-      // Update the todoList state with the new array of todos
-      setTodoList(updatedTodoList);
+
+      // Transform API data.records into todo objects
+      const todos = data.records.map((record) => ({
+        id: record.id,
+        title: record.fields.title,
+      }));
+
+
+      // Update todoList and isLoading states
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(`Fetch error: ${error.message}`);
     }
+  }
 
 
-    // Handle adding a new todo
-    function addTodo(newTodo) {
-      setTodoList([...todoList, newTodo]); // Update todoList state by adding the new todo
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+
+  // Use useEffect to manage localStorage and persist todoList
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList)); // Save the todoList in localStorage when it changes
     }
+  }, [todoList, isLoading]);
+
+
+  // Handle adding a new todo
+  async function handleAddTodo(newTodo) {
+    try {
+      const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          records: [{ fields: { title: newTodo.title } }],
+        }),
+      });
+
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
 
-    return (
-      <>
-        <AddTodoForm onAddTodo={addTodo} />
-        {isLoading ? <p>Loading...</p> : <TodoList todoList={todoList} onRemoveTodo={removeTodo} />}
-        
 
-      </>
-    );
+      const data = await response.json();
+
+
+      // Update the todoList state by adding the new todo
+      setTodoList([...todoList, { id: data.id, title: newTodo.title }]);
```

Comment:
There's a small hidden bug here, which is that `data` is still an object with an array, and that array contains the actual todo that was created: 

<img width="761" alt="Screen Shot 2023-08-24 at 12 09 06 AM" src="https://github.com/Victoria240/react-todo/assets/12771651/447651b1-d24d-4a23-af9d-a793307462d0">

Reply:
so to actually get the id and title confirmed by the server, you'll have to "unwrap" the layers of data around the todo: 

```suggestion
      const data = await response.json();

      const confirmedTodo = data.records[0];

      // Update the todoList state by adding the new todo
      setTodoList([...todoList, { id: confirmedTodo.id, title: confirmedTodo.fields.title }]);
```

Reply:
I also want to point call attention to changing `newTodo.title` to `confirmedTodo.fields.title` - 

Because the backend is supposed to be the "source of truth" for all of our data, we should trust the title returned from Airtable over the title that we sent to Airtable, just in case they somehow become different. It won't make a difference in this case, but it's good to keep in mind for the future!